packages/headless/src/controllers/search-parameter-manager/headless-search-parameter-manager.ts

```diff
@@ -209,9 +209,8 @@ function getNumberOfResults(state: Partial<SearchParametersState>) {
     return {};
   }
 
-  const numberOfResults = state.pagination.numberOfResults;
-  const shouldInclude =
-    numberOfResults !== getPaginationInitialState().numberOfResults;
+  const {numberOfResults, defaultNumberOfResults} = state.pagination;
```

Comment:
I decided to name this `defaultNumberOfResults` instead of `initialNumberOfResults` to mimic `advancedSearchQueries`'s `defaultFilters`.